Shared/mods/deathmatch/logic/lua/LuaBasic.h

```diff
@@ -228,12 +228,25 @@ namespace lua
         return 1;
     }
 
-    // Tuples can be used to return multiple results
+    // Tuples can be used to build fixed-size tables
+    // e.g. `std::tuple<float, int, bool>` will be pushed as { float, int, bool }
     template<typename... Ts>
     int Push(lua_State* L, const std::tuple<Ts...>& tuple)
     {
         // Call Push on each element of the tuple
-        std::apply([L](const auto&... value) { (Push(L, value), ...); }, tuple);
-        return sizeof...(Ts);
+        lua_createtable(L, sizeof...(Ts), 0);
+        
+        std::apply([L](const auto&... values) {
+            int  key = 1;
+            auto PushTable = [](lua_State* L, int& key, const auto& value)
+            {
+                Push(L, key++);
+                Push(L, value);
+                lua_settable(L, -3);
```

Comment:
`lua_rawsetfield` is more efficient as it doesn't try to call the tag method, also it doesn't need the key on the stack.
```cpp
Push(L, value);
lua_rawsetfield(L, -3, key++);
```

Reply:
Where did you get `lua_rawsetfield` from? That's not a function in Lua 5.1

Reply:
I'm sorry. Its [lua_rawseti](https://pgl.yoyo.org/luai/i/lua_rawseti).